.github/workflows/iroha2-dev-nightly.yml

```diff
@@ -3,17 +3,26 @@ name: I2::Dev::Nightly::Publish
 on: workflow_dispatch
 
 env:
+  RUST_TOOLCHAIN: nightly-2024-04-18
   DOCKER_COMPOSE_PATH: defaults
 
 jobs:
   build_executor:
-    runs-on: ubuntu-latest
-    container:
-      image: hyperledger/iroha2-ci:nightly-2024-04-18
+    runs-on: macos-latest
```

Comment:
what was the reason for the switch? also, we don't get to use `mold` anymore

Reply:
Building on M1 should be faster, and Apple's `ld-prime` is as good as `mold`